src/Connection/Http1Connection.php

```diff
@@ -445,8 +456,15 @@ private function readResponse(
                 \strlen($parser->getBuffer()),
                 getCurrentTime() - $start
             ));
-        } catch (StreamException $e) {
-            throw new SocketException('Receiving the response headers failed: ' . $e->getMessage());
+        } catch (HttpException $e) {
+            $this->close();
+            throw $e;
+        } catch (PromiseTimeoutException $e) {
+            $this->close();
+            throw new TimeoutException('Receiving the response timed out due to inactivity', 0, $e);
```

Comment:
The exception message should also have the timeout in the message here.

Reply:
Both locations fixed.